services/slam/builtin/builtin.go

```diff
@@ -368,7 +370,32 @@ func (slamSvc *builtIn) Position(ctx context.Context, name string) (*referencefr
 		return nil, errors.Wrap(err, "error getting SLAM position")
 	}
 
-	return referenceframe.ProtobufToPoseInFrame(resp.Pose), nil
+	pInFrame := referenceframe.ProtobufToPoseInFrame(resp.Pose)
+
+	// TODO DATA-531: Remove extraction and conversion of quaternion from the extra field in the response once the Rust SDK
+	// is available and the desired math can be implemented on the orbSLAM side
+	extra := resp.Extra.AsMap()
+
+	if val, ok := extra["quat"]; ok {
+		q := val.(map[string]interface{})
+
+		valTheta, ok1 := q["otheta"]
+		valX, ok2 := q["ox"]
+		valY, ok3 := q["oy"]
+		valZ, ok4 := q["oz"]
+
+		if !ok1 || !ok2 || !ok3 || !ok4 {
+			slamSvc.logger.Warnf("invalid format for quaternion returned, %v, skipping quaternion transform", q)
+			return pInFrame, nil
+		}
+		slamSvc.logger.Infof("valid format for quaternion returned, %v, performing quaternion transform", q)
```

Comment:
Logging of position has been added, however depending on the frequency at which GetPosition is called by the UI this may be removed in the future.

Reply:
missed that there was logging in both rdk and orbslam, probably only want to log in one of them if you know the grpc call is working

Reply:
If you leave this in, I would make it a debug instead of info.

Reply:
@JohnN193 I'll plan on keeping the logging here and will remove from orbslam